articles/watch-your-state.md

```diff
@@ -0,0 +1,49 @@
+
+#### State in Today's Communication Protocols
+
+Turning from industrial production to Internet communication protocols, the goals of improving quality and reducing costs translate to reducing latency and lowering energy consumption:
+
+1. When Internet user Alice connects to her online bank, she needs to authenticate before accessing her savings and transactions. To initiate such an authenticated session, she provides unique credentials (login and password, and hopefully a second authentication factor). If the login is successful, the bank grants Alice an access token valid for that very session. Alice can then seamlessly re-authenticate herself during the session using that token.
+
+    Such a token contains information about the session, in particular, information to identify the user associated with the session. The token thus represents session *state*. The question we ask is where and how the bank should manage such temporary access tokens? The bank could store a copy of millions of such session tokens and perform equality checks with the tokens that Alice sends with her requests. If Alice's token exists in the list of copied tokens, the bank can safely assume it's indeed Alice who connects.
+
+    This, however, is very inefficient as the bank's server needs to perform several memory lookups for each customer request to fetch the corresponding token. Unfortunately, most web application firewalls (WAFs) today work exactly this way.
+
+    Before discussing smarter ways of handling access tokens, we look at two more examples.
+
+
+2. Packet forwarding in today's Internet: When Alice sends a data packet to Bob, she adds Bob's IP address to the packet header and hands the packet over to her Internet provider. From then on, every router on the path from Alice to Bob extracts Bob's address from the packet to do a database lookup to retrieve information about the next hop, that is, where to forward the packet to. Each router bases its forwarding decision on local *state* that was previously learned from neighboring routers.
+
+    This procedure is not only insecure (as the neighbors' information is not properly authenticated), but also very energy-consuming and inefficient (as millions of packets per second need to be looked up in specialized parallelizable hardware called TCAMs). Good news: A more secure and -- at the same time -- more efficient solution exists.
```

Comment:
(as millions of packets per second need to be looked up in specialized parallelizable hardware called TCAMs) -> (as millions of packets per second need to be looked up) // kürzungspotential

Reply:
hm... ja... Die versteckte Aussage sollte sein, dass trotz der hochspezialisierten Hardware das Vorgehen noch immer ineffizient ist. Aber ja, ich kürze! :)